src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -64,14 +71,7 @@ export default class ResultsViewPathwayMapper extends React.Component<
         onResult: (genes: string[]) => {
             // show loading text only if there are actually new genes to load
             if (genes.length > 0) {
-                const tId = toast('Loading alteration data...', {
-                    autoClose: false,
-                    draggable: false,
-                    position: 'bottom-left',
-                    className: styles.toast,
-                });
-
-                this.activeToasts.push(tId);
+                this.showMessage(this.LOADING_MESSAGE);
```

Comment:
instead of setting observable state here, we should just put conditions in render which test validNonQueryGenes.isComplete && validNonQueryGnes.length > 0.     The fundamental idea is not to make state redundant but computed as much as possible.

Reply:
since we are not dealing with problematic toasts anymore, it is worth trying to get rid of the `onResult` callback and handle the conditions in `render`.

Reply:
Wouldn't it cause infinite render loop if I set the message inside the render function?

Reply:
@Rgtemze It would. You shouldn't set an observable neither here nor in the renderer. Instead, as @alisman suggested, you should test the condition `validNonQueryGenes.isComplete && validNonQueryGnes.length > 0` in renderer and decide which message to show there.